vale/specs/crypto/Vale.AES.AES_BE_s.fst

```diff
@@ -0,0 +1,157 @@
+module Vale.AES.AES_BE_s
+
+// IMPORTANT: This specification is written assuming a big-endian mapping from bytes to quad32s
+//            This is explicit in key_schedule_to_round_keys when we construct the round_key rk,
+//            but it also applies implicitly to the input quad32
+
+open Vale.Def.Prop_s
+open Vale.Def.Opaque_s
+open Vale.Def.Words_s
+open Vale.Def.Words.Four_s
+open Vale.Def.Words.Seq_s
+open Vale.Def.Types_s
+open Vale.Arch.Types
+open FStar.Seq
+open FStar.Mul
+
+// substitution is endian-neutral;
+// others operations assume that the quad32 and nat32 values are big-endian interpretations
+// of 16-byte and 4-byte sequences
+assume val mix_columns (q:quad32) : quad32
+assume val inv_mix_columns (q:quad32) : quad32
+assume val sub_bytes (q:quad32) : quad32
+assume val inv_sub_bytes (q:quad32) : quad32
+assume val shift_rows (q:quad32) : quad32
+assume val inv_shift_rows (q:quad32) : quad32
+assume val rot_word (w:nat32) : nat32
+assume val sub_word (w:nat32) : nat32
+
+assume val commute_rot_word_sub_word (x:nat32) (rcon:nat32) : Lemma
+  (rot_word (sub_word x *^ ishl32 rcon 16) == sub_word (rot_word x) *^ rcon)
+
+type algorithm:eqtype = | AES_128 | AES_192 | AES_256
```

Comment:
Many definitions in this module seem copy-pasted from the Vale.AES.AES_s. Could we merge the two files in one, using _LE/_BE suffixes when there is a difference? If not possible, could we have common definitions (e.g., algorithm type) in a single file?

Reply:
Done!